task/net/src/bsp/gimletlet_mgmt.rs

```diff
@@ -0,0 +1,223 @@
+        //  TXD1 is on a different pin)
+        //
+        //  The MDIO/MDC lines run at Speed::Low because otherwise the VSC8504
+        //  refuses to talk.
+        use gpio_api::*;
+        let gpio = Gpio::from(GPIO.get_task_id());
+        let eth_af = Alternate::AF11;
+
+        // RMII
+        gpio.configure(
+            Port::A,
+            (1 << 1) | (1 << 7),
+            Mode::Alternate,
+            OutputType::PushPull,
+            Speed::VeryHigh,
+            Pull::None,
+            eth_af,
+        )
+        .unwrap();
+        gpio.configure(
+            Port::C,
+            (1 << 4) | (1 << 5),
+            Mode::Alternate,
+            OutputType::PushPull,
+            Speed::VeryHigh,
+            Pull::None,
+            eth_af,
+        )
+        .unwrap();
+        gpio.configure(
+            Port::G,
+            (1 << 11) | (1 << 12) | (1 << 13),
+            Mode::Alternate,
+            OutputType::PushPull,
+            Speed::VeryHigh,
+            Pull::None,
+            eth_af,
+        )
+        .unwrap();
+
+        // SMI (MDC and MDIO)
+        gpio.configure(
+            Port::A,
+            1 << 2,
+            Mode::Alternate,
+            OutputType::PushPull,
+            Speed::Low,
+            Pull::None,
+            eth_af,
+        )
+        .unwrap();
+        gpio.configure(
+            Port::C,
+            1 << 1,
+            Mode::Alternate,
+            OutputType::PushPull,
+            Speed::Low,
+            Pull::None,
+            eth_af,
+        )
+        .unwrap();
+    }
+
+    pub fn configure_phy(&self, eth: &mut eth::Ethernet) {
+        // The KSZ8463 connects to the SP over RMII, then sends data to the
+        // VSC8552 over 100-BASE FX
+        self.ksz.configure();
+
+        // The VSC8552 connects the KSZ switch to the management network
+        // over SGMII
+        configure_vsc8552(eth);
+    }
+
+    pub fn wake(&self, eth: &mut eth::Ethernet) {
+        let p1_sr = self.ksz.read(KszRegister::P1MBSR).unwrap();
+        ringbuf_entry!(Trace::Ksz8463Status {
+            port: 1,
+            status: p1_sr
+        });
+
+        let p2_sr = self.ksz.read(KszRegister::P2MBSR).unwrap();
+        ringbuf_entry!(Trace::Ksz8463Status {
+            port: 2,
+            status: p2_sr
+        });
+
+        for i in [0, 1] {
+            let port = VSC8552_PORT + i;
+            let status = eth.smi_read(port, eth::SmiClause22Register::Status);
+            ringbuf_entry!(Trace::Vsc8552Status { port, status });
+
+            let status = eth
+                .smi_read(port, eth::SmiClause22Register::TxFxExtendedStatus);
+            ringbuf_entry!(Trace::Vsc8552Status100 { port, status });
+        }
+    }
+}
+
+////////////////////////////////////////////////////////////////////////////////
+
+/// Helper struct to implement the `PhyRw` trait using direct access through
+/// `eth`'s MIIM registers.
+struct MiimBridge<'a> {
```

Comment:
This looks identical to the one in sidecar_1?

Reply:
Yup, deduplicated and moved to a single place.